### ISSUE TITLE
"Go to Definition/Declaration" in file with UNC path opens new tab

### ISSUE THREAD

**Issue by Niko-O** [microsoft/vscode-cpptools#2528]
**Type: LanguageService**

**Describe the bug**
- OS and Version: Windows 10 (10.0.17134.285) 64 bit
- VS Code Version:
```
	Version 1.14.2
	Commit cb82febafda0c8c199b9201ad274e25d9a76874e
	Datum 2017-07-19T23:34:09.706Z
	Shell 1.6.6
	Renderer 56.0.2924.87
	Node 7.4.0
```
- C/C++ Extension Version: 0.14.0
- Other extensions you installed (and if the issue persists after disabling them):
  - hexdump for VSCode
  - Intel HEX format
  - macros
  - Yes, the issue persists after disabling all extensions (except for `C/C++`, of course).
- A clear and concise description of what the bug is:
When using the "Go to Definition" or "Go to Declaration" commands in a `.h`, `.c` or `.cpp` file, if the file where the declaration/definition resides is opened with a UNC path (e.g. something like `\\A\B\C\D.E`), then the first part of the path to that file (`\\A\B`) is converted to lower case before opening. For example: If the target file is `\\Bar\Foo\Baz\Test.c`, then the file `\\bar\foo\Baz\Test.c\ will be opened instead.
This causes a new tab to appear in most cases, even though both file paths, in practice, point to the same file.

**To Reproduce**
1. In Windows Explorer, create a new folder called `Foo` anywhere you have access to (e.g. on your Desktop).
2. Right click on the folder, select "Properties".
3. In the tab "Sharing" click on "Advanced Sharing".
4. Enable "Share this folder" and click "OK".
5. Note the path shown below "Network path" in the properties dialog and click OK.
6. Create a new folder called `Baz` in `Foo`.
7. Create a new file called `Test.c` in `Baz` and insert this code:
```
int x = 0;
void y()
{
    x = 1;
}
```
8. Open a new instance of Visual Studio Code.
9. In the menu "File" click on "Open Folder".
10. In the address bar of the dialog, type the UNC path to `Foo` as noted above, hit Enter and click "Select Folder".
11. In the explorer panel, expand `Baz` and double click on `Test.c`.
12. Right click on `x` in line 4 and click on "Go to Declaration".
13. Observe that, even though the declaration is in the currently opened file, a new tab is opened.
14. Hover over the tab headers to see the full paths. The already open file reads `\\Bar\Foo\Baz\Test.c` while the newly opened file reads `\\bar\foo\Baz\Test.c`.

**Expected behavior**
After executing step 12, the editor should have simply scrolled to line 1 in the same file. It should not have opened a separate tab.

**Screenshots**
![share](https://user-images.githubusercontent.com/4943909/45685184-d3305680-bb48-11e8-8844-ed5ecab42665.png)

![Paths](https://user-images.githubusercontent.com/4943909/45684812-bb0c0780-bb47-11e8-84c3-72c5e330e0d8.png)

**Additional context**
It appears that, according to the change log, there is a newer version of the extension available (0.18.1), but Visual Studio Code does not show a button which would allow me to update.
![update](https://user-images.githubusercontent.com/4943909/45686005-7c784c00-bb4b-11e8-9cc2-b64adcd8663e.png)

**Comment by sean-mcmanus** [comment]
VS Code has a known problem with opening different files on Windows if the casing is incorrect (https://github.com/Microsoft/vscode/issues/12448 ). We have code to retrieve the file system casing from Windows, but that seems to be buggy with UNC paths.

**Comment by Niko-O** [comment]
I see. Would it be possible to reuse the opened folder's path as-is? That is, "Go to Declaration" would use `\\Bar\Foo\Baz\Test.c` if the currently opened folder is `\\Bar\Foo` and likewise `\\bar\foo\Baz\Test.c` if the opened path is `\\bar\foo` instead of asking Windows to deduce the casing.

**Comment by sean-mcmanus** [comment]
Yeah, we could probably do something like that to fix it...we haven't investigated it yet.

**Comment by michelleangela** [comment]
Fixed in version 0.26.2-insiders, available at https://github.com/microsoft/vscode-cpptools/releases/tag/0.26.2-insiders.
